allauth/account/forms.py

```diff
@@ -335,13 +336,23 @@ def __init__(self, *args, **kwargs):
         )
 
     def clean_username(self):
-        value = self.cleaned_data["username"]
-        value = get_adapter().clean_username(value)
+        if self.cleaned_data["username"]:
+            value = self.cleaned_data["username"]
+        else:
+            value = get_adapter().clean_username(value)
```

Comment:
This looks wrong -- `value` is passed to `clean_username()` without being assigned a value first.

Reply:
That makes sense. Except I am not really sure what to do here. 

If left the way it was originally like this:
```
        if self.cleaned_data["username"]:
            value = self.cleaned_data["username"]
            value = get_adapter().clean_username(value)
```
the "A link to activate your account has been emailed to the address provided" would not execute and the default "User with this username already exists." will be used instead. 

That is because I am not using an adapter. The value variable gets passed into the query `User.objects.filter(username__iexact=value)` so there needs to be a way to determine if an adapter is being used versus if an adapter is not being used.